File: C/Algorithms/2006.c
#define COUNT_MAX 101  // Max value in nums + 1.

int countKDifference(int* nums, int numsSize, int k){
    // normally no, due to constraints ok for this case.
    int counter[COUNT_MAX] = {0};
    
    for (int i = 0; i < numsSize; i++)
        counter[nums[i]]++;
    
    int count = 0;
    for (int i = 0; i < numsSize; i++) {
        int num = nums[i];
        int a = num - k, b = num + k;
        if (a > 0 && a < COUNT_MAX && counter[a])
            count += counter[a];
        if (b < COUNT_MAX && counter[b])
            count += counter[b];
        counter[num]--;        
    }
    return count;
}

File: Go/Algorithms/2006.go
package main

func countKDifference(nums []int, k int) int {
	counts := make(map[int]int)

	for _, i := range nums {
		counts[i]++
	}

	count := 0
	for _, i := range nums {
		a, b := i-k, i+k
		if a > 0 && a < 101 && counts[a] > 0 {
			count += counts[a]
		}
		if b < 101 && counts[b] > 0 {
			count += counts[b]
		}
		counts[i]--
	}

	return count
}
